src/ui/pages/ConnectionDetails/ConnectionDetails.tsx

```diff
@@ -138,12 +138,19 @@ const ConnectionDetails = () => {
 
   const verifyAction = () => {
     async function deleteConnection() {
+      if (!connectionDetails) return;
       await Agent.agent.connections.deleteConnectionById(
         connectionShortDetails.id
       );
-      const updatedConnections = connectionsData.filter(
-        (item) => item.id !== connectionDetails?.id
+
+      const newConnectionsData = Object.fromEntries(
```

Comment:
Not sure on this way of deleting. Can we not just do `delete connectionDetails[connectionShortDetails.id]`?

Reply:
@iFergal _connectionShortDetails.id_ is a _non-configurable_ property of connectionDetails . That means we can't delete it

Reply:
Sorry, I meant data - `delete connectionsData[connectionShortDetails.id]` - this should delete the key-value entry in the object so we can update redux more clearly (and more efficiently)

Reply:
I tried but it didn't work
![image](https://github.com/user-attachments/assets/6fe514ef-9bea-4542-93ed-21a15a823213)